bin/stt_test/stt_providers/ISTTBase.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Literal, Protocol, TypedDict
+import speech_recognition as sr  # type: ignore
+import time
+from stt_test.utils.SentenceChecker import SentenceChecker
+from stt_test.utils.test_data import get_audio_with_transcription
+import inspect
+
+
+class STTResult(TypedDict):
+    stt_trans: str
+    trans: str
+    similarity: float
+    execution_time: float
+
+
+class ISSTBase(Protocol):
+
+    MODEL_SIZES: List[str]
+    _sc: SentenceChecker
+    _recognizer: sr.Recognizer
+
+    def _init(self):
+        self._sc = SentenceChecker()
+        self._recognizer = sr.Recognizer()
+
+    def _before_all(self, model_size: str):
+        """Method for preparing models"""
+        ...
+
+    def _audio_to_text(self, audio: sr.AudioData, language: str) -> str:
+        """Methods that runs STT on a single audio"""
+        ...
+
+    def _save_results(self, data: List[STTResult], model: str):
+        stack = inspect.stack()
+        caller_class = stack[1][0].f_locals["self"].__class__.__name__
+        with open(f'report_{caller_class}_{model}.json', 'w') as fp:
+            json.dump(data, fp)
+
+    def run_analysis(self, file: str, trans: str, language: str):
+        with sr.AudioFile(file) as source:
+            audio = self._recognizer.record(source)  # type: ignore
+        try:
+            start_time = time.time()
+            stt_trans = self._audio_to_text(audio, language)
+            execution_time = time.time() - start_time
+            similarity = self._sc.check_similarity(stt_trans, trans)
+            return STTResult(stt_trans=stt_trans, trans=trans, similarity=similarity,
+                             execution_time=execution_time)
+        except sr.UnknownValueError:
```

Comment:
We could print the message of the error as well.

Reply:
We can, but is there any point since the exception will print this error anyway?

Reply:
I mean, it will ease the debugging if we print the message from UnkownValueError here, or at least if we pass it to the exception that is being thrown here.